src/nuclio/functions/version/version-triggers/version-triggers.component.js

```diff
@@ -237,10 +217,48 @@
             });
         }
 
+        /**
+         * Tests whether the `HTTP trigger message` is shown
+         * @returns {boolean} `true` in case "HTTP trigger message" is shown, or `false` otherwise.
+         */
+        function isHttpTriggerMsgShown() {
+            var httpTrigger = lodash.find(ctrl.version.spec.triggers, ['kind', 'http'])
+
+            return lodash.isNil(httpTrigger)
+        }
+
         //
         // Private methods
         //
 
+        /**
+         * Creates the new trigger item
+         * @param {Object} [trigger] - trigger data from `ctrl.version`
+         * @returns {Object} - trigger item
+         */
+        function createTriggerItem(trigger) {
+            var triggerItem = lodash.assign(trigger ? trigger : {}, {
```

Comment:
1. I'm not sure I get what you meant to do here.
`lodash.assign` mutates its first argument.
Do you intend to change the `trigger` you receive as an argument (in case it _is_ provided)?
If not, you should add `{}` as first argument in order not to mutate `trigger`:
```js
var triggerItem = lodash.assign({}, trigger ? trigger : {}, {
//                              ^^^^
```

2. You repeatedly used `lodash.isNil(trigger)` later in this function.
How about having some `var noTriggerProvided = lodash.isNil(trigger)` in the beginning, and then reuse this boolean, for example instead of `trigger ? …` use `noTriggerProvided ? …`, and instead of `if (!trigger)` use `if (noTriggerProvided)`.